src/miner.cpp

```diff
@@ -60,13 +60,12 @@ BlockAssembler::Options::Options() {
     nBlockMaxSize = DEFAULT_BLOCK_MAX_SIZE;
 }
 
-BlockAssembler::BlockAssembler(const CSporkManager& sporkManager, CGovernanceManager& governanceManager,
-                               LLMQContext& llmq_ctx, CEvoDB& evoDb, CChainState& chainstate, const CTxMemPool& mempool, const CChainParams& params, const Options& options) :
+BlockAssembler::BlockAssembler(LLMQContext& llmq_ctx, MNSubsidyAgent& mn_subsidy, CEvoDB& evoDb, CChainState& chainstate,
```

Comment:
`mn_sybsidy` is a part of `chainstate` now; why do we need to pass it explicitly?

Reply:
For the same reason we pass `mempool` and `params` , the references/pointers passed to `CChainState` [are `protected`](https://github.com/dashpay/dash/blob/015e30fa0be7b6bcd0654285085e5883e4587a07/src/validation.h#L655-L669). We can access `m_chain` because [it's `public`](https://github.com/dashpay/dash/blob/015e30fa0be7b6bcd0654285085e5883e4587a07/src/validation.h#L707-L709).